isisd/isis_pdu.c

```diff
@@ -532,7 +530,7 @@ static int process_lan_hello(struct iih_info *iih)
 		}
 	}
 
-	adj->circuit_t = iih->circ_type;
+	adj->circuit_t = iih->calculated_type;
```

Comment:
Why does this need to be in a separate commit?

Reply:
thanks for your comments, i have corrected all your requirements

Reply:
> Why does this need to be in a separate commit?

thanks for your comments, i have corrected all your requirements now for sure